java-checks/src/main/java/org/sonar/java/checks/ConstantMathCheck.java

```diff
@@ -0,0 +1,195 @@
+        if (isIntegralOne(remainderTree.rightOperand())) {
+          addIssue(tree, "Remove this computation of % 1, which always evaluates to zero.");
+        }
+      } else {
+        MethodInvocationTree methodTree = (MethodInvocationTree) tree;
+        if (isTruncation(methodTree) || isConstantWithZero(methodTree) || isConstantWithZeroOrOne(methodTree)) {
+          addIssue(tree, String.format("Remove this silly call to \"Math.%s\"", methodTree.symbol().name()));
+        }
+      }
+    }
+  }
+
+  private boolean isTruncation(MethodInvocationTree methodTree) {
+    return TRUNCATION_METHODS.anyMatch(methodTree) && isCastFromIntegralToFloating(removeParenthesis(methodTree.arguments().get(0)));
+  }
+
+  private boolean isConstantWithZero(MethodInvocationTree methodTree) {
+    return CONSTANT_WITH_ZERO_METHODS.anyMatch(methodTree) && isFloatingZero(methodTree.arguments().get(0));
+  }
+
+  private boolean isConstantWithZeroOrOne(MethodInvocationTree methodTree) {
+    return CONSTANT_WITH_ZERO_OR_ONE_METHODS.anyMatch(methodTree) && isFloatingZeroOrOne(methodTree.arguments().get(0));
+  }
+
+  private boolean isCastFromIntegralToFloating(ExpressionTree tree) {
+    Type resultType = tree.symbolType();
+    // explicit cast
+    if (tree.is(Tree.Kind.TYPE_CAST) && isIntegral(getInnerType(((TypeCastTree) tree).expression())) && (resultType.is(DOUBLE) || resultType.is(FLOAT))) {
+      return true;
+    }
+    // implicit cast
+    return isIntegral(resultType);
+  }
+
+  private boolean isIntegral(Type type) {
+    return type.isPrimitive() && !type.is(DOUBLE) && !type.is(FLOAT);
+  }
+
+  private boolean isIntegralOne(ExpressionTree tree) {
+    Long value = LiteralUtils.longLiteralValue(tree);
+    return value != null && value == 1;
+  }
+
+  private Type getInnerType(ExpressionTree tree) {
+    ExpressionTree result = tree;
+    while (true) {
+      if (result.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+        result = ((ParenthesizedTree) result).expression();
+      } else if (result.is(Tree.Kind.TYPE_CAST)) {
+        result = ((TypeCastTree) result).expression();
+      } else {
+        return result.symbolType();
+      }
+    }
+  }
+
+  private ExpressionTree removeParenthesis(ExpressionTree tree) {
+    ExpressionTree result = tree;
+    while (result.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
+      result = ((ParenthesizedTree) result).expression();
+    }
+    return result;
+  }
+
+  private boolean isFloatingZero(ExpressionTree tree) {
+    Double value = parseDouble(removeParenthesis(tree));
+    return value != null && Double.compare(value, 0.0d) == 0;
+  }
+
+  private boolean isFloatingZeroOrOne(ExpressionTree tree) {
+    // current implementation of parseDouble returns null for all value but 0 and 1, so there is no need to check the actual returned value.
+    return parseDouble(removeParenthesis(tree)) != null;
+  }
+
+  @CheckForNull
+  private Double parseDouble(ExpressionTree tree) {
```

Comment:
The method name should be changed to match what it currently does. As you said it, It only returns `1.0`, `0.0` or `null` but not parse any double from the provided expression.